### PR TITLE
Add magma build for CUDA 11

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Copied from 10.2 with additional changes.

- cmakelists.patch:
  - add sm80 and compute80
  - disable magma sparse
- magma/magma-cuda110/cudaPointerAttributes.patch

### CODE REVIEW

File: magma/magma-cuda110/cmakelists.patch [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+ 
+     # nvcc >= 6.5 supports -std=c++11, so propagate CXXFLAGS to NVCCFLAGS.
+     # Older nvcc didn't support -std=c++11, so previously we disabled propagation.
+@@ -259,11 +261,18 @@
+         message( STATUS "    compile for CUDA arch 7.5 (Turing)" )
+     endif()
+ 
++    if ( ${GPU_TARGET} MATCHES "All")
++      set( MIN_ARCH 370)
++      SET( NV_SM -gencode arch=compute_37,code=sm_37 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_70,code=sm_70 )
```
Comment (malfet) [inline magma/magma-cuda110/cmakelists.patch:61]: Hmm, don't you want to add compute_80 here?

File: magma/magma-cuda110/cmakelists.patch [hunk 2]
```diff
@@ -0,0 +1,332 @@
+--- CMakeLists.txt	2020-06-18 14:34:00.713576314 -0700
++++ CMakeLists.txt	2020-06-18 14:36:23.821607272 -0700
+@@ -3,7 +3,7 @@
+ # ----------------------------------------
+ # to disable Fortran, set this to "off"
+ # see also -DADD_ below
+-option( USE_FORTRAN "Fortran is required for some tester checks, but can be disabled with reduced functionality" ON )
++option( USE_FORTRAN "Fortran is required for some tester checks, but can be disabled with reduced functionality" OFF )
```
Comment (malfet) [inline magma/magma-cuda110/cmakelists.patch:8]: Why this change is needed, if cmake is called with `-DUSE_FORTRAN=OFF`?

File: magma/magma-cuda110/cmakelists.patch (hunk 1 above)
Comment (zasdfgbnm) [reply magma/magma-cuda110/cmakelists.patch]: Good catch! I remembered I added it here but looks like I was wrong. Fixed now.

File: magma/magma-cuda110/cmakelists.patch (hunk 2 above)
Comment (zasdfgbnm) [reply magma/magma-cuda110/cmakelists.patch:8]: I don't know, this part is copied from CUDA 10.2 patch.

**Comment by dawonn-haval** [comment]
Any guess when this might be merged?

**Comment by zasdfgbnm** [comment]
@dawonn-haval I believe it will be merged soon after 1.6 release of pytorch
